src/plugins/mer/meremulatordetailswidget.cpp

```diff
@@ -215,5 +227,14 @@ void MerEmulatorDetailsWidget::selectFactorySnapshot()
     emit factorySnapshotChanged(selected);
 }
 
+void MerEmulatorDetailsWidget::updateSshPortAvailableLabel()
+{
+    const bool vmOff = (m_emulator == nullptr
+                        || m_emulator->virtualMachine() == nullptr
+                        || m_emulator->virtualMachine()->isOff());
+    const bool available = MerLocalhostAvailablePorts::isAvailable(m_ui->sshPortSpinBox->value());
+    m_ui->sshPortAvailableLabel->setHidden(available || !vmOff);
+}
+
```

Comment:
Ditto. As with build engine, the logic should be moved to `MerEmulatorOptionsWidget`.

Reply:
Ok.

Reply:
This is done. Do we need check ssh port is available by timer when Settings window is open?

Reply:
I think it is enough to check it on change.